src/storage/mvcc/txn.rs

```diff
@@ -482,6 +484,16 @@ impl<S: Snapshot> MvccTxn<S> {
             .into());
         }
         if let Some((commit_ts, _)) = self.reader.seek_write(key, TimeStamp::max())? {
+            // The invariants of pessimistic locks are:
+            //   1. lock's for_update_ts > key's latest commit_ts
+            //   2. lock's for_update_ts >= txn's start_ts
+            //   3. If the data is changed, key's new commit_ts > lock's for_update_ts
```

Comment:
Seems 1 and 3 contradict, is 1 true except when handling new data?

Reply:
They are not contradict, but the comments are not clear. If succeeds in acquiring pessimistic locks, 1 is true. If the data is changed after acquiring pessimistic locks(dut to pipelining procedure), 3 is true.